components/datadog/dogstatsd-standalone/k8s.go

```diff
@@ -0,0 +1,258 @@
+
+	k8sComponent := &K8sComponent{}
+	if err := e.Ctx.RegisterComponentResource("dd:dogstatsd-standalone", "dogstatsd", k8sComponent, opts...); err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(k8sComponent))
+
+	ns, err := corev1.NewNamespace(
+		e.Ctx,
+		namespace,
+		&corev1.NamespaceArgs{
+			Metadata: metav1.ObjectMetaArgs{
+				Name: pulumi.String(namespace),
+			},
+		},
+		opts...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, utils.PulumiDependsOn(ns))
+
+	daemonSetArgs := appsv1.DaemonSetArgs{
+		Metadata: &metav1.ObjectMetaArgs{
+			Name:      pulumi.String("dogstatsd-standalone"),
+			Namespace: pulumi.String(namespace),
+		},
+		Spec: &appsv1.DaemonSetSpecArgs{
+			Selector: &metav1.LabelSelectorArgs{
+				MatchLabels: pulumi.StringMap{
+					"app": pulumi.String("dogstatsd-standalone"),
+				},
+			},
+			Template: &corev1.PodTemplateSpecArgs{
+				Metadata: &metav1.ObjectMetaArgs{
+					Labels: pulumi.StringMap{
+						"app": pulumi.String("dogstatsd-standalone"),
+					},
+				},
+				Spec: &corev1.PodSpecArgs{
+					HostPID:            pulumi.BoolPtr(true),
+					ServiceAccountName: pulumi.String("dogstatsd-standalone"),
+					Containers: corev1.ContainerArray{
+						&corev1.ContainerArgs{
+							Name:  pulumi.String("dogstatsd-standalone"),
+							Image: pulumi.String(e.DogstatsdFullImagePath()),
+							Ports: corev1.ContainerPortArray{
+								&corev1.ContainerPortArgs{
+									ContainerPort: pulumi.Int(8125),
+									HostPort:      pulumi.Int(HostPort),
+									Name:          pulumi.StringPtr("dogstatsdport"),
+									Protocol:      pulumi.StringPtr("UDP"),
+								},
+							},
+							Env: &corev1.EnvVarArray{
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_API_KEY"),
+									Value: e.DogstatsdAPIKey(),
+								},
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_LOG_LEVEL"),
+									Value: pulumi.StringPtr("DEBUG"),
+								},
+								&corev1.EnvVarArgs{
+									Name: pulumi.String("DD_KUBERNETES_KUBELET_HOST"),
+									ValueFrom: corev1.EnvVarSourceArgs{
+										FieldRef: corev1.ObjectFieldSelectorArgs{
+											FieldPath: pulumi.String("status.hostIP"),
+										},
+									},
+								},
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_DOGSTATSD_NON_LOCAL_TRAFFIC"),
+									Value: pulumi.StringPtr("true"),
+								},
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_DOGSTATSD_ORIGIN_DETECTION"),
+									Value: pulumi.StringPtr("true"),
+								},
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_DOGSTATSD_SOCKET"),
+									Value: pulumi.StringPtr(Socket),
+								},
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_DOGSTATSD_TAG_CARDINALITY"),
+									Value: pulumi.StringPtr("orchestrator"),
+								},
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_KUBELET_TLS_VERIFY"),
+									Value: pulumi.StringPtr("false"),
+								},
```

Comment:
Is this necessary ?
[I couldn’t see `kubelet.tlsVerify: false` for the agent](https://github.com/DataDog/test-infra-definitions/blob/8158a5b2c60e9d31f827390168a916a2534c2ac1/components/datadog/agent/helm.go).

Reply:
It doesn't work for me with tls verify set to true, but I'm not sure why it works in the agent and not dogstatsd standalone.

Reply:
Actually, when deploying on Kind we also set tlsVerify: false for the agent:
https://github.com/DataDog/test-infra-definitions/blob/8158a5b2c60e9d31f827390168a916a2534c2ac1/scenarios/aws/kindvm/run.go#L58

Reply:
Ah ! Thanks for catching it !
So, for `dogstatsd-standalone`, would it be possible to also have `kubelet:tlsVerify: false` only for Kind and not for all kubernetes setups ?